### PR TITLE
Update pool interface

### PR DESCRIPTION
**Pull request by trgill**
Add parameters to the pool interface
Correct the name of the pool interface
Change create_dbus_pool to take a reference
Remove un-needed clone of dbus_context

### CODE REVIEW

**Comment by trgill** [comment]
@mulkieran @agrover this look ok?

**Comment by agrover** [comment]
looks good, except please fix the commitmsg for 3f3dc75 : since the first line is treated as a summary line let's do it right - omit the dash and fix the typo. With that fixed you get my :+1:

**Comment by trgill** [comment]
fixed commit message typo

**Comment by agrover** [comment]
:+1:

File: src/dbus_api.rs [hunk 1]
```diff
@@ -185,35 +183,68 @@ fn remove_devs(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
...
+    let create_filesystems_method = f.method(CREATE_FILESYSTEMS, (), create_filesystems)
+        .in_arg(("volumes", "a(sss)"))
+        .out_arg(("results", "a(oqs)"))
+        .out_arg(("return_code", "q"))
+        .out_arg(("return_string", "s"));
 
-    let destroy_filesystems_method = f.method(DESTROY_FILESYSTEMS, (), destroy_filesystems);
+    let destroy_filesystems_method = f.method(DESTROY_FILESYSTEMS, (), destroy_filesystems)
+        .in_arg(("volumes", "a(sqs)"))
+        .out_arg(("results", "a(oqs)"))
```
Comment (mulkieran) [inline src/dbus_api.rs]: Let's change results to "a(qs)" since the object path is now obsolete and we get the analogy of returning a just-freed pointer.

**Pull request merged**